flake.nix

```diff
@@ -413,10 +449,15 @@
             ctl-unit-test = project.runPursTest {
               name = "ctl-unit-test";
               testMain = "Test.Unit";
-            };
+            }.overrideAttrs
+              (oas: {
+                checkPhase = exportOgmiosFixtures + oas.checkPhase;
+              });
           };
 
-          devShell = project.devShell;
+          devShell = project.devShell.overrideAttrs (oas: {
+            shellHook = oas.shellHook + exportOgmiosFixtures;
```

Comment:
Since you moved `buildOgmiosFixtures` from `nix/default.nix` you don't need to do an override anymore. This can be passed directly to the `shell` argument of `purescriptProject`, so starting at line 411, it should be 

```
            shell = {
              shellHook = exportOgmiosFixtures;
              packages = [
               # ...
              ];
            };
``` 
and this can be reverted to 
```
           devShell = project.devShell;
```